data/lib/src/network/model/response/upload_request_response.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:json_annotation/json_annotation.dart';
+
+part 'upload_request_response.g.dart';
+
+@JsonSerializable()
+@DatetimeConverter()
+@UploadRequestIdConverter()
+class UploadRequestResponse extends Equatable {
+  UploadRequestResponse(
+      this.uploadRequestId,
+      this.owner,
+      this.recipients,
+      this.activationDate,
+      this.modificationDate,
+      this.creationDate,
+      this.expiryDate,
+      this.notificationDate,
+      this.label,
+      this.status,
+      this.maxFileCount,
+      this.maxDepositSize,
+      this.maxFileSize,
+      this.canDeleteDocument,
+      this.canClose,
+      this.body,
+      this.pristine,
+      this.protectedByPassword,
+      this.extensions,
+      this.enableNotification,
+      this.canEditExpiryDate,
+      this.collective,
+      this.nbrUploadedFiles
+  );
+
+  @JsonKey(name: Attribute.uuid)
+  final UploadRequestId uploadRequestId;
+
+  final UploadRequestAccountDto owner;
+  final Set<UploadRequestAccountDto> recipients;
+  final DateTime activationDate;
+  final DateTime modificationDate;
+  final DateTime creationDate;
+
+  // could be null
+  final DateTime expiryDate;
+
+  final DateTime notificationDate;
+  final String label;
+  final UploadRequestStatus status;
+  final int maxFileCount;
+
+  // could be null
+  final double maxDepositSize;
+
+  // could be null
+  final double maxFileSize;
+
+  final bool canDeleteDocument;
+  final bool canClose;
+
+  // could be null
+  final String body;
+
+  final bool isClosed = false;
+  final bool pristine;
+  final bool protectedByPassword;
+  final double usedSpace = 0;
+  final Set<String> extensions;
+  final bool enableNotification;
+  final bool canEditExpiryDate;
+  final bool collective;
+  final int nbrUploadedFiles;
+
+  factory UploadRequestResponse.fromJson(Map<String, dynamic> json) =>
+      _$UploadRequestResponseFromJson(json);
+  Map<String, dynamic> toJson() => _$UploadRequestResponseToJson(this);
+
+  @override
+  List<Object> get props => [
+      uploadRequestId,
+      owner,
+      recipients,
+      activationDate,
+      modificationDate,
+      creationDate,
+      expiryDate,
+      notificationDate,
+      label,
+      status,
+      maxFileCount,
+      maxDepositSize,
+      maxFileSize,
+      canDeleteDocument,
+      canClose,
+      body,
+      pristine,
+      protectedByPassword,
+      extensions,
+      enableNotification,
+      canEditExpiryDate,
+      collective,
+      nbrUploadedFiles
+  ];
+}
+
+extension UploadRequestResponseExtension on UploadRequestResponse {
+  UploadRequest toUploadRequest() {
+    return UploadRequest(
+      uploadRequestId,
+      owner.toUploadRequestAccount(),
+      recipients.map((recipient) => recipient.toUploadRequestAccount()).toSet(),
+      activationDate,
+      modificationDate,
+      creationDate,
+      expiryDate,
+      notificationDate,
+      label,
+      status,
+      maxFileCount,
+      maxDepositSize,
```

Comment:
Because it could be null, it should have an unexpected/default value here. Such as: `maxDepositSize ?? 0`. The same to others

Reply:
You are right, what default value I should put for dateTime ?

Reply:
I think it should be a constant DateTime, a special time, such as: 01/01/2001. It may help us to debug faster afterward